rskj-core/src/test/java/co/rsk/peg/federation/NonStandardErpFederationsTest.java

```diff
@@ -611,7 +605,7 @@ void createErpFederation_mainnet_constants_after_RSKIP293() {
     @Test
     void getRedeemScript_before_RSKIP_284_testnet() {
         networkParameters = NetworkParameters.fromID(NetworkParameters.ID_TESTNET);
-        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        activations = mock(ActivationConfig.ForBlock.class);
```

Comment:
do we need to mock this again? is not already done in the `setup()` ?

Reply:
It's basically overriding the mock created in the setup

Reply:
but why mocking it again?

Reply:
To override the original object

Reply:
Since the setup is doing this `when(activations.isActive(ConsensusRule.RSKIP201)).thenReturn(true);`, it needs to be overridden if you want to have other specific RSKIPs mocked.

Reply:
but there's no need to mock the whole object again. We just have to mock the response to the rskip284.

Reply:
but there's no need to mock the whole object again right? We just have to mock the response to the rskip284

Reply:
The thing is, the old mock object may have mock things that you are not interested in.

Reply:
we should have the rskip201 activated in all the tests (thats why is being activated in the setup). We are not mocking the whole object again in any of the other tests, just adding the mocked responses we need.

Reply:
If that's the case, you are right, no need for creating a new instance.

Reply:
updated